src/aatoolbox/datasources/iri/iri_forecast_seasonal_precipitation_tercile.py

```diff
@@ -0,0 +1,184 @@
+"""Class to download and load IRI's seasonal forecast.
+
+Data is downloaded from `IRI's maproom
+<https://iridl.ldeo.columbia.edu/maproom/Global/Forecasts/NMME_Seasonal_Forecasts/Precipitation_ELR.html>`_
+
+For now only the tercile precipitation forecast has been
+implemented.
+"""
+
+import logging
+
+import requests
+import xarray as xr
+
+import aatoolbox.utils.raster  # noqa: F401
+from aatoolbox.datasources.datasource import DataSource
+from aatoolbox.utils.geoboundingbox import GeoBoundingBox
+
+logger = logging.getLogger(__name__)
+
+_MODULE_BASENAME = "iri"
+
+
+class IriForecast(DataSource):
+    """
+    Work with IRI's forecast data.
+
+    Parameters
+    ----------
+    iso3 : str
+        country iso3
+    geo_bounding_box: GeoBoundingBox
+        the bounding coordinates of the area that should
+        be included in the data.
+    """
+
+    def __init__(self, iso3, geo_bounding_box: GeoBoundingBox):
+        super().__init__(
+            iso3=iso3, module_base_dir=_MODULE_BASENAME, is_public=False
+        )
+        self._iso3 = iso3
+        # round coordinates to correspond with the grid IRI publishes
+        # its data on, which is 1 degree resolution
+        # non-rounded coordinates can be given to the URL which then
+        # automatically rounds them, but for file saving we prefer to do
+        # this ourselves
+        geo_bounding_box.round_coords(round_val=1, offset_val=0)
+        self._geobb = geo_bounding_box
+
+    def download(
+        self,
+        iri_auth: str,
+        # question: I chose to make dominant a function arg
+        # instead of class attr. Does this make sense?
+        # Does it even make sense to have dominant as an arg
+        # or should we have a separate class for it?
```

Comment:
Hm yeah that's a tough call. I don't really understand what the parameter is, is it it something where in most cases you would want to download both dominant and the alternative? You could write down the workflow for either case and see what you think makes the most sense.

Btw for booleans I think it's good to start them with a verb, e.g. `use_dominant` or `is_dominant`.

Reply:
:) I am trying this out. But then I would say the "main class" doesn't have to be a class anymore. Just a generalized download, process, and load function. Would you agree?

Reply:
>  Just a generalized download, process, and load function. Would you agree? 

Hm not sure, don't these methods need some of the instance variables?